pandas/io/parsers.py

```diff
@@ -801,6 +803,28 @@ def _get_options_with_defaults(self, engine):
 
         return options
 
+    def _check_file_or_buffer(self, f, engine):
+        # see gh-16530
+        if is_file_like(f):
+            next_attr = "__next__" if PY3 else "next"
+
+            # The C engine doesn't need the file-like to have the "next" or
+            # "__next__" attribute. However, the Python engine explicitly calls
+            # "next(...)" when iterating through such an object, meaning it
+            # needs to have that attribute ("next" for Python 2.x, "__next__"
+            # for Python 3.x)
+            if engine != "c" and not hasattr(f, next_attr):
+                msg = ("The 'python' engine cannot iterate "
+                       "through this file buffer.")
+                if self._engine_specified:
+                    raise ValueError(msg)
+                else:
+                    engine = "c"
+                    msg += " Falling back to the 'c' engine."
+                    warnings.warn(msg, ParserWarning, stacklevel=5)
```

Comment:
Appears this isn't being hit by test coverage.  Should the warning be checked?  Can this actually be hit?